src/auth/src/credentials/external_account_sources/executable_sourced.rs

```diff
@@ -0,0 +1,317 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{
+    Result,
+    constants::{ACCESS_TOKEN_TYPE, JWT_TOKEN_TYPE, SAML2_TOKEN_TYPE},
+    credentials::external_account::SubjectTokenProvider,
+};
+use gax::error::CredentialsError;
+use serde::{Deserialize, Serialize};
+use std::time::Duration;
+use tokio::{process::Command, time::timeout as tokio_timeout};
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub(crate) struct ExecutableSourcedCredentials {
+    pub executable: ExecutableConfig,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Default)]
+pub(crate) struct ExecutableConfig {
+    pub command: Option<String>,
+    pub timeout_millis: Option<u32>,
+    pub output_file: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+struct ExecutableResponse {
+    version: i32,
+    success: bool,
+    token_type: String,
+    expiration_time: i64,
```

Comment:
This is only used in tests, and it is unclear what the units are. Are these seconds from the epoch? Milliseconds from the epoch? Something else?